### PR TITLE
RAB: Integrate staging tests for the .filter method

### PR DESCRIPTION
**Pull request by ioannad**
of Array.prototype and TypedArray.prototype

This is part of PR https://github.com/tc39/test262/pull/3888 to make reviewing easier. Includes changes to use the helper ./harness/resizableArrayBufferUtils.js

### CODE REVIEW

File: test/built-ins/Array/prototype/filter/resizable-buffer.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-array.prototype.filter
+description: >
+  Array.p.filter behaves correctly on receivers backed by resizable
+  buffers
+includes: [compareArray.js, resizableArrayBufferUtils.js]
+features: [resizable-arraybuffer]
+flags: [onlyStrict]
```
Comment (ptomato) [inline test/built-ins/Array/prototype/filter/resizable-buffer.js:11]: Do we need `onlyStrict` or is it leftover from the staging file?

**Review by ptomato — APPROVED** [review]
Was there a `Array/prototype/filter/resizable-buffer-shrink-mid-iteration.js` in the original PR? It's a bit odd that there's only the TypedArray version in this PR.

Otherwise looks fine, possibly remove the `noStrict` flag.

File: test/built-ins/Array/prototype/filter/resizable-buffer.js (hunk 1 above)
Comment (ioannad) [reply test/built-ins/Array/prototype/filter/resizable-buffer.js]: Done.

**Comment by ioannad** [comment]
> Was there a `Array/prototype/filter/resizable-buffer-shrink-mid-iteration.js` in the original PR? It's a bit odd that there's only the TypedArray version in this PR.
> 
> Otherwise looks fine, possibly remove the `noStrict` flag.

@ptomato  Not sure, I'll check tomorrow in more detail and change the PR label when I'm done.

**Comment by ioannad** [comment]
@ptomato:

 > Was there a `Array/prototype/filter/resizable-buffer-shrink-mid-iteration.js` in the original PR? It's a bit odd that there's only the TypedArray version in this PR.

There was not, neither in #3888 not any tests in staging for `Array.prototype.filter` when shrinking mid-iteration. Do you think this should be tested as well, either here or in a follow-up PR adding missing tests?
 
> Otherwise looks fine, possibly remove the `noStrict` flag.

Removed.

**Comment by ptomato** [comment]
Shrinking mid-iteration with filter is fine to add in a followup. I wouldn't consider it that high of a priority, it's just odd that it's tested everywhere else and not here.

**Pull request merged**